code/datums/components/singularity.dm

```diff
@@ -61,6 +62,8 @@
 	src.notify_admins = notify_admins
 	src.roaming = roaming
 	src.singularity_size = singularity_size
+	if(immediate_target)
+		target = immediate_target
```

Comment:
You're not hooking to when this is deleted, this is going to hard delete. *That is why it was a weakref in the first place*. Weakrefs are for non-owning references. This is a non-owning reference.

Reply:
This was a hard reference to the target, I changed nothing about target handling except allowing it to be passed in addcomponent
This part is your mistake but id be willing to fix it